ownCloud/Client/Viewer/Media/MediaDisplayViewController.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  MediaDisplayViewController.swift
+//  ownCloud
+//
+//  Created by Michael Neuwert on 30.06.2019.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+/*
+* Copyright (C) 2018, ownCloud GmbH.
+*
+* This code is covered by the GNU Public License Version 3.
+*
+* For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+* You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+*
+*/
+
+import UIKit
+import AVKit
+import ownCloudSDK
+
+class MediaDisplayViewController : DisplayViewController {
+
+	private var playerStatusObservation: NSKeyValueObservation?
+	private var playerItemStatusObservation: NSKeyValueObservation?
+	private var playerItem: AVPlayerItem?
+	private var player: AVPlayer?
+
+	deinit {
+		playerStatusObservation?.invalidate()
+		playerItemStatusObservation?.invalidate()
+	}
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+		self.requiresLocalItemCopy = false
+	}
+
+	override func renderSpecificView() {
+		if source != nil {
+
+			let asset = AVURLAsset(url: source!, options: self.httpAuthHeaders != nil ? ["AVURLAssetHTTPHeaderFieldsKey" : self.httpAuthHeaders!] : nil )
```

Comment:
if you use `if let source = source` before, you can remove force unwrapping here

Reply:
@hosy fixed in 74ec9adc270a144663aa045e6bc8ddded8583f1b